pkg/provider/aws/execution_unit.go

```diff
@@ -12,6 +12,25 @@ import (
 	"go.uber.org/zap"
 )
 
+// expandExecutionUnit takes in a single execution unit and expands the generic construct into a set of resource's based on the units configuration.
+func (a *AWS) expandExecutionUnit(dag *core.ResourceGraph, unit *core.ExecutionUnit) error {
+	switch a.Config.GetExecutionUnit(unit.ID).Type {
+	case Lambda:
+		lambda, err := core.CreateResource[*resources.LambdaFunction](dag, resources.LambdaCreateParams{
+			AppName: a.Config.AppName,
+			Refs:    []core.AnnotationKey{unit.AnnotationKey},
```

Comment:
Won't lambda always have exactly 1 Ref?

Reply:
yeah it should right now, but maybe not always if we allow people to combine units or something. Also i tried to keep the create params consistent if possible